continuationsPlugin/src/test/scala/continuations/ContinuationsPluginSuite.scala

```diff
@@ -706,4 +702,99 @@ class ContinuationsPluginSuite extends FunSuite, CompilerFixtures {
+           |        protected override def invokeSuspend(
+           |          result: Either[Throwable, Any | Null | continuations.Continuation.State.Suspended.type]
+           |        ): Any | Null = 
+           |          {
+           |            this.$result = result
+           |            this.$label = this.$label | Int.MinValue
+           |            foo(this.asInstanceOf[Continuation[Int]])
+           |          }
+           |    }
+           |
+           |    def foo(completion: Continuation[Int]): Int | Null | Continuation.State.Suspended.type = {
+           |      var $continuation: Continuation[Any] | Null = null
+           |
+           |      completion match {
+           |        case continuations$foo$ : continuations$foo$1
+           |            if (continuations$foo$.$label & Int.MinValue) != 0x0 =>
+           |          $continuation = continuations$foo$
+           |
+           |          $continuation.asInstanceOf[continuations$foo$1].$label =
+           |            $continuation.asInstanceOf[continuations$foo$1].$label - Int.MinValue
+           |
+           |        case _ =>
+           |          $continuation = new continuations$foo$1(
+           |            completion.asInstanceOf[Continuation[Any | Null]])
+           |      }
+           |
+           |      val $result: Either[Throwable, Any | Null | Continuation.State.Suspended.type] =
+           |        $continuation.asInstanceOf[continuations$foo$1].$result
+           |
+           |      $continuation.asInstanceOf[continuations$foo$1].$label match
+           |        case 0 =>
+           |          $result.fold (t => throw t, _ => ())
+           |
+           |          $continuation.asInstanceOf[continuations$foo$1].$label = 1
+           |
+           |          val safeContinuation: continuations.SafeContinuation[Int] =
+           |            new continuations.SafeContinuation[Int](continuations.intrinsics.IntrinsicsJvm$package.intercepted[Int]($continuation)(),
+           |              continuations.Continuation.State.Undecided
+           |            )
+           |          safeContinuation.resume(Right(1))
+           |
+           |          val orThrow: Any | Null | Continuation.State.Suspended.type = safeContinuation.getOrThrow()
+           |          if (orThrow == Continuation.State.Suspended) {
+           |            return Continuation.State.Suspended
+           |          }
+           |        case 1 =>
+           |          $result.fold(t => throw t, _ => ())
+           |        case _ =>
+           |          throw new IllegalStateException ("call to 'resume' before 'invoke' with coroutine")
+           |      10
+           |    }
+           |  }
+           |}
+           |""".stripMargin
+      // format: on
+
+      checkCompile("lambdaLift", source) {
+        case (tree, _) =>
+          println(tree.show)
+//          assertNoDiff(compileSourceIdentifier.replaceAllIn(tree.show, ""), expected)
```

Comment:
Probably need to replace the assertion now, right?

Reply:
yes, although I broke some existing code around detecting the def that needs to change so will have to fix this back first